### PR TITLE
[sw, bazel] Build and bazel target fixes for off-target (unit test) coverage

### PR DESCRIPTION
**Pull request by alphan**
This PR fixes several build issues to be able to build silicon creator code with `clang`. The last commit removes a superfluous dependency from silicon creator targets for more accurate coverage reports.

### CODE REVIEW

File: sw/device/lib/base/memory_unittest.cc [hunk 1]
```diff
@@ -71,12 +86,12 @@ class MemSetTest : public ::testing::TestWithParam<decltype(&ot_memset)> {};
 class MemChrTest : public ::testing::TestWithParam<decltype(&ot_memchr)> {};
 
 INSTANTIATE_TEST_SUITE_P(MemCpy, MemCpyTest,
-                         ::testing::Values(ot_memcpy, __builtin_memcpy));
+                         ::testing::Values(ot_memcpy, builtin_memcpy_wrapper));
```
Comment (dmcardle) [inline sw/device/lib/base/memory_unittest.cc:89]: Interesting, do you remember what the error was?

**Review by dmcardle — APPROVED** [review]
LGTM!

File: sw/device/lib/base/memory_unittest.cc (hunk 1 above)
Comment (alphan) [reply sw/device/lib/base/memory_unittest.cc:89]: Yes, it was `builtin functions must be directly called`

Comment (alphan) [reply sw/device/lib/base/memory_unittest.cc:89]: relevant [thread](https://gcc.gnu.org/legacy-ml/gcc-bugs/2015-06/msg01141.html) (which is pretty old)

**Comment by alphan** [comment]
Thanks for the quick review @dmcardle !

**Pull request merged**